lib/minidoc/finders.rb

```diff
@@ -24,17 +24,16 @@ def exists?(selector = {})
 
     def find(id_or_selector, options = {})
       if id_or_selector.is_a?(Hash)
-        options.merge!(transformer: method(:wrap))
-        collection.find(id_or_selector, options)
+        collection.find(id_or_selector, options).lazy.map(&method(:wrap))
```

Comment:
You're right that `sort` and `count` use the `Array` implementation and cause database queries. Most other enumerable methods have lazy implementations, though: http://ruby-doc.org/core-2.0.0/Enumerator/Lazy.html

and in my testing, they behave correctly (they don't issue database queries)